core/src/main/java/org/mskcc/cbio/portal/servlet/MutationsJSON.java

```diff
@@ -118,41 +132,48 @@ private void processGetSmgRequest(HttpServletRequest request,
         Map<Long, Map<String, String>> smgs = Collections.emptyMap();
         DaoGeneOptimized daoGeneOptimized = DaoGeneOptimized.getInstance();
         try {
-            mutationProfile = DaoGeneticProfile.getGeneticProfileByStableId(mutationProfileId);
-            if (mutationProfile!=null) {
-                int profileId = mutationProfile.getGeneticProfileId();
-                List<Integer> selectedCaseList = new ArrayList<Integer>();
-                
-                if (request.getParameterMap().containsKey("case_list")) {
-                    String caseList = request.getParameter("case_list");
-                    Pattern p = Pattern.compile("[,\\s]+");
-                    String sampleIds[] = p.split(caseList);
-                    CancerStudy cancerStudy = DaoCancerStudy.getCancerStudyByInternalId(mutationProfile.getCancerStudyId());
-                    selectedCaseList = InternalIdUtil.getInternalSampleIds(cancerStudy.getInternalId(), Arrays.asList(sampleIds));
-                }
-                
-                // get all recurrently mutation genes
-                smgs = DaoMutation.getSMGs(profileId, null, 2, DEFAULT_THERSHOLD_NUM_SMGS, (selectedCaseList.isEmpty()? null: selectedCaseList));
-                
-                // get all cbio cancer genes
-                Set<Long> cbioCancerGeneIds = daoGeneOptimized.getEntrezGeneIds(
-                        daoGeneOptimized.getCbioCancerGenes());
-                cbioCancerGeneIds.removeAll(smgs.keySet());
-                if (!cbioCancerGeneIds.isEmpty()) {
-                    smgs.putAll(DaoMutation.getSMGs(profileId, cbioCancerGeneIds, -1, -1, (selectedCaseList.isEmpty()? null: selectedCaseList)));
-                }
-                
-                // added mutsig results
-                mutsig = getMutSig(mutationProfile.getCancerStudyId());
-                if (!mutsig.isEmpty()) {
-                    Set<Long> mutsigGenes = new HashSet<Long>(mutsig.keySet());
-                    mutsigGenes.removeAll(smgs.keySet());
-                    if (!mutsigGenes.isEmpty()) {
-                        // append mutsig genes
-                        smgs.putAll(DaoMutation.getSMGs(profileId, mutsigGenes, -1, -1, (selectedCaseList.isEmpty()? null: selectedCaseList)));
-                    }
-                }
-            }
+			if (mutationProfileId != null) {
+				// Get the Genetic Profile
+				mutationProfile = DaoGeneticProfile.getGeneticProfileByStableId(mutationProfileId);
+				if (mutationProfile != null) {
+					CancerStudy cancerStudy = DaoCancerStudy
```

Comment:
Why not check once for data access in processRequest instead of all the individual routines?